mabwiser/linear.py

```diff
@@ -90,50 +90,47 @@ def predict(self, x):
         # Calculate default expectation y = x * b
         return np.dot(x, self.beta)
 
-    def _scale_predict_context(self, x):
+    def _scale_predict_context(self, x: np.ndarray):
         if not hasattr(self.scaler, 'scale_'):
             return x
 
-        # Reshape 1D array to 2D
-        x = x.reshape(1, -1)
-
         # Transform and return to previous shape. Convert to float64 to suppress any type warnings.
-        return self.scaler.transform(x.astype('float64')).reshape(-1)
+        return self.scaler.transform(x.astype('float64'))
 
 
 class _LinTS(_RidgeRegression):
 
-    def predict(self, x):
-
+    def predict(self, x: np.ndarray):
         # Scale
         if self.scaler is not None:
             x = self._scale_predict_context(x)
 
         # Randomly sample coefficients from multivariate normal distribution
         # Covariance is enhanced with the exploration factor
-        beta_sampled = self.rng.multivariate_normal(self.beta, np.square(self.alpha) * self.A_inv)
+        beta_sampled = self.rng.multivariate_normal(self.beta, np.square(self.alpha) * self.A_inv, size=x.shape[0])
 
         # Calculate expectation y = x * beta_sampled
-        return np.dot(x, beta_sampled)
+        return np.sum(x * beta_sampled, axis=1)
 
 
 class _LinUCB(_RidgeRegression):
 
-    def predict(self, x):
-
+    def predict(self, x: np.ndarray):
         # Scale
         if self.scaler is not None:
             x = self._scale_predict_context(x)
 
+        # Calculating x_A_inv
+        x_A_inv = np.dot(x, self.A_inv)
+
         # Upper confidence bound = alpha * sqrt(x A^-1 xt). Notice that, x = xt
-        ucb = (self.alpha * np.sqrt(np.dot(np.dot(x, self.A_inv), x)))
+        ucb = self.alpha * np.sqrt(np.sum(x_A_inv * x, axis=1))
```

Comment:
faster?

Reply:
Nope. This change was done to accomodate multiple inputs/contexts at once and calulate the ucb value respectively. Now both the input- x and -the variable ucb are of np.ndarray type